connection.go

```diff
@@ -505,10 +505,16 @@ func (c *Connection) Reply(ctx context.Context, opErr error) {
 	noResponse := c.kernelResponse(outMsg, inMsg.Header().Unique, op, opErr)
 
 	if !noResponse {
-		err := c.writeMessage(outMsg.Bytes())
+		var err error
+		if outMsg.Sglist != nil {
```

Comment:
Could we alternatively move the initialization code from `Append()` somewhere else so that all messages have an Sglist that contains at least the header? Then, we don’t need to distinguish here and can remove the `Bytes()` method entirely

Reply:
I'm not sure if it really matters, but the idea here was to avoid an extra allocation of slice :-)